src/fw/node_modules/fwrule/index.js

```diff
@@ -192,14 +193,6 @@ parser.yy.parseError = function parseError(_, details) {
         throw err;
     }
 
-    if (details.text === '') {
-        err = new validators.InvalidParamError('rule',
-            'Error at character 0: \'\', expected: \'FROM\', found: '
-            + 'empty string');
-        err.details = details;
-        throw err;
-    }
-
```

Comment:
Same explanation (updated sdc- here as in sdc-fwrule#5, right?

Reply:
Yep, exactly the same. SmartOS just has its own full copy.